### PR TITLE
Azure pipelines setup

### PR DESCRIPTION
**Pull request by rth**
Working on setting up CI with Azure Pipelines

### CODE REVIEW

**Comment by rth** [comment]
So there are several issues while trying tests,
 - Python 2.7 the dependencies (and in particular numba) won't install, because of version mismatch https://github.com/rtavenar/tslearn/issues/191 I'm not sure it's worth spending time on this, dropping python 2.7 might be a solution.
 - Python 3.5: some tests are very slow in CI. Locally tests run for me in a minute or so, while in CI some individual tests timeout after 90s. I believe this is related to TensorFlow, maybe due to CPU oversubscription: in other CI the detected number of CPU used to be much larger than the number of available CPU i.e  2. This is usually addressed by setting environement variables, but it looks like [Tensorflow 1.15  dropped support for OMP_NUM_THREADS](https://github.com/tensorflow/tensorflow/releases/tag/v1.15.0). This needs more investigation, meanwhile maybe 

I'm planning to add other OSs once Linux test pass.

**Comment by rtavenar** [comment]
Hi @rth 

Thanks for this. Yeah I think for the moment you can move forward with Python 2 disabled and skip `ShapeletModel`-related tests. I can try to have a deeper look at the specific problem related to TF.

**Comment by GillesVandewiele** [comment]
Some of the tests in check_estimators (especially those with ShapeletModels as @rtavenar pointed out) are indeed rather slow. This is because the default hyper-parameters of `ShapeletModel` will have a long runtime (e.g. max_iter of 10000...). Perhaps we should look to extend it with early stopping or smth like that, as it converges quickly on the benchmark datasets.

**Comment by rth** [comment]
>  This is because the default hyper-parameters of ShapeletModel will have a long runtime (e.g. max_iter of 10000...).

Thanks! I see, maybe it could be worth setting it to a lower value in tests similarly to what is done in https://github.com/scikit-learn/scikit-learn/blob/c3969f43d9e4ba66d7f211d3835bb4cbe616c12b/sklearn/utils/estimator_checks.py#L481

For now I will skip them.

**Comment by rth** [comment]
OK so I skipped ShapeletModel tests for now and all tests run in less than 4 min. We can add wheel building (and Python 3.8 https://github.com/rtavenar/tslearn/issues/193) in a follow up PR. Current setup is a bit repetitive but it's a start.

**Comment by GillesVandewiele** [comment]
That is a great idea, and that reminded me that we actually tried to do something similar:
https://github.com/rtavenar/tslearn/blob/master/tslearn/tests/test_estimators.py#L168

But it seems to not be working as it should?

**Comment by rth** [comment]
> that reminded me that we actually tried to do something similar:

Indeed, I opened https://github.com/rtavenar/tslearn/issues/194 to discuss how to do that without redefining check_estimator, but generally yes that should have worked.

**Comment by rtavenar** [comment]
OK, so the nice thing (correct me if I'm wrong) is that here you discarded Shapelet tests only for Azure pipelines, so we can still use Travis in parallel for the moment until we have found a fix for Shapelet timeouts, right?

If this is true, I would merge and we can have a look at that #194 you opened.

File: tslearn/tests/test_utils.py [hunk 1]
```diff
@@ -46,6 +48,7 @@ def test_label_categorizer():
 
 
 def test_conversions():
+    pytest.importorskip('pandas')
```
Comment (rtavenar) [inline tslearn/tests/test_utils.py]: Hum, I have a concern here: I would prefer that we skipped only pandas-related conversion functions (ie. those involving sktime, pyflux and tsfresh) only. Maybe we could split the tests into two functions (one with pandas required, one that should pass even without pandas)?

File: tslearn/utils.py [hunk 2]
```diff
@@ -770,471 +770,483 @@ def transpose_or_expand(ts):
     return to_time_series_dataset([transpose_or_expand(Xi) for Xi in X])
 
 
-try:  # Ugly hack, not sure how to to it better
-    import pandas as pd
+def to_sktime_dataset(X):
+    """Transform a tslearn-compatible dataset into a sktime dataset.
 
-    def to_sktime_dataset(X):
```
Comment (rth) [inline tslearn/utils.py]: Having the `try: ... except ImportError:` inside the functions is a more standard way of doing it. Done that and removed 1 indentation from all of these functions.

**Comment by rth** [comment]
>  would prefer that we skipped only pandas-related conversion functions (ie. those involving sktime, pyflux and tsfresh) only. Maybe we could split the tests into two functions (one with pandas required, one that should pass even without pandas)?

OK, I split that test in multiple functions depending on dependencies.

I would be good to merge this, if only to avoid CI failures in other PRs.

File: tslearn/utils.py (hunk 2 above)
Comment (rtavenar) [reply tslearn/utils.py]: OK, but now it seems that on Travis doctests fail if the dependencies are not properly installed.

Comment (rtavenar) [reply tslearn/utils.py]: The question is: why do these pass on Azure? Are doctests executed at all?

Comment (rth) [reply tslearn/utils.py]: I saw, working on a fix. No I need to add the corresponding pytest option.

**Comment by rth** [comment]
OK, I added a ` conftest.py` to skip doctests that require optional dependencies (as it is done in scikit-learn). Hopefully that would fix Travis.

For now Azure doesn't run doctests indeed, I would rather add it in a follow up PR. For one thing, it would also be good to move the `doc/` folder one level up, to avoid excluding it from tests.

**Comment by rth** [comment]
What's the difference between dev and master branches BTW? This is a PR into master, should it be done to dev instead?. I guess the naming of branches doesn't matter much, but it would be good to have the default branch on github to be the one in active development, so that PRs are made to the correct one..

**Comment by rtavenar** [comment]
> What's the difference between dev and master branches BTW? This is a PR into master, should it be done to dev instead?. I guess the naming of branches doesn't matter much, but it would be good to have the default branch on github to be the one in active development, so that PRs are made to the correct one..

You're right. What we have right now is:

* `master` is the branch from which wheels, conda packages, and docs are built
* `dev` is a branch used to prepare next release (contains new features to appear, etc)

**Pull request merged**